packages/utils/tests/bucket.spec.ts

```diff
@@ -169,5 +169,28 @@ describe('bucket.ts', () => {
       expect(bucket.remaining).equals(0)
       expect(bucket.used).equals(1)
     })
+
+    it("remaining is 0 when used too many", () => {
+      const bucket = new LeakyBucket({
+        max: 1,
+        refillInterval: 500,
+        refillAmount: 1,
+      })
+      // max is < used
+      bucket.used = 2;
+      expect(bucket.remaining).equals(0);
+    })
+
+    it("Don't process queue twice", () => {
+      const bucket = new LeakyBucket({
+        max: 1,
+        refillInterval: 500,
+        refillAmount: 1,
+      })
+      // fake processing
+      bucket.processing = true;
+      // request when already processing
+      bucket.processQueue();
```

Comment:
missing a check here

Reply:
what check?

Reply:
how do you check if it processes twice?

Reply:
it doesn't process twice, the second request when already processing is true should just cancel out, no way to check that